// demo.proto
syntax = "proto3";

message DataMessage {
  int32 id = 1;                     // Unique ID for the message from the client
  string payload = 2;
  repeated int32 values = 3;
  int64 timestamp_emission = 4;     // Client-side emission timestamp (e.g., epoch nanoseconds)
}

message Ack {
  string message = 1;
}

service GRPCDemo {
  rpc StreamData(stream DataMessage) returns (Ack);
}